### PR TITLE
Update managed auth docs for automatic refresh improvement.

### PR DESCRIPTION
**Pull request by ThumulaPerera**
## Purpose
$subject

### CODE REVIEW

File: en/docs/authentication-and-authorization/secure-web-applications-with-managed-authentication.md [hunk 1]
```diff
@@ -115,23 +115,11 @@ If you enable Choreo's managed authentication, you don't have to manually add an
...
-
-If the refresh token is valid, the session will be refreshed, and the system will respond with a `204 No Content` status.
-
-If the refresh token has expired, the session will not be refreshed, and the system will respond with a `401 Unauthorized` status.
-
-To automate the session refresh process on receiving a `401 unauthorized` response from the Choreo API, you can encapsulate the requests with the refresh logic. The following is an example code snippet that wraps GET requests:
+To programmatically re-login on receiving a `401 unauthorized` response from the Choreo API, you can encapsulate the requests with the re-login logic. The following is an example code snippet that wraps GET requests:
 
 ``` javascript
     export const performGetWithRetry = async (url) => {
```
Comment (senthalan) [inline en/docs/authentication-and-authorization/secure-web-applications-with-managed-authentication.md:125]: We need to update this method name.

The developer doesn't know about the automatic token refresh.

File: en/docs/authentication-and-authorization/secure-web-applications-with-managed-authentication.md [hunk 2]
```diff
@@ -115,23 +115,11 @@ If you enable Choreo's managed authentication, you don't have to manually add an
 If Choreo's managed authentication is disabled, you must ensure that your web application attaches a valid access token to the API call.
 
 
-### Step 1.5: Refresh session
+### Step 1.5: Handle session expiry
 
-The logged is user's session expires when the access token provided by the identity provider you have configured expires. After the session expires, the session must be refreshed to successfully call a Choreo-deployed API. For session refresh to function, you must set up the identity provider to issue a refresh token in addition to the access token.
+The logged is user's session expires when the time configured for `Session Expiry Time` is exceeded. `401 Unauthorized` response status code for a Choreo API request from a logged in user with the necessary permissions indicates a possible session expiry. After the session expires, the user should re-login.
```
Comment (Suvink) [inline en/docs/authentication-and-authorization/secure-web-applications-with-managed-authentication.md:120]:
```suggestion
The logged in user's session expires when the time configured for `Session Expiry Time` is exceeded. A `401 Unauthorized` response status code for a Choreo API request from a logged in user with the necessary permissions indicates a possible session expiry. After the session expires, the user should re-login.
```

File: en/docs/authentication-and-authorization/secure-web-applications-with-managed-authentication.md [hunk 3]
```diff
@@ -115,23 +115,11 @@ If you enable Choreo's managed authentication, you don't have to manually add an
...
-``` javascript
-const response = await fetch("/auth/refresh", { method: "POST" })
-```
-
-If the refresh token is valid, the session will be refreshed, and the system will respond with a `204 No Content` status.
-
-If the refresh token has expired, the session will not be refreshed, and the system will respond with a `401 Unauthorized` status.
-
-To automate the session refresh process on receiving a `401 unauthorized` response from the Choreo API, you can encapsulate the requests with the refresh logic. The following is an example code snippet that wraps GET requests:
+To programmatically re-login on receiving a `401 unauthorized` response from the Choreo API, you can encapsulate the requests with the re-login logic. The following is an example code snippet that wraps GET requests:
```
Comment (Suvink) [inline en/docs/authentication-and-authorization/secure-web-applications-with-managed-authentication.md:122]:
```suggestion
To programmatically re-login on receiving a `401 unauthorized` response from a Choreo API, you can encapsulate the request with the re-login logic. The following is an example code snippet that wraps GET requests:
```

Comment (ThumulaPerera) [reply en/docs/authentication-and-authorization/secure-web-applications-with-managed-authentication.md:122]: Resolved with [Address comments](https://github.com/wso2/docs-choreo-dev/pull/1542/commits/fe955e7e298df1cd4770bf6e2158341b6ef6eed4)

File: en/docs/authentication-and-authorization/secure-web-applications-with-managed-authentication.md (hunk 2 above)
Comment (ThumulaPerera) [reply en/docs/authentication-and-authorization/secure-web-applications-with-managed-authentication.md]: Resolved with [Address comments](https://github.com/wso2/docs-choreo-dev/pull/1542/commits/fe955e7e298df1cd4770bf6e2158341b6ef6eed4)

File: en/docs/authentication-and-authorization/secure-web-applications-with-managed-authentication.md (hunk 1 above)
Comment (ThumulaPerera) [reply en/docs/authentication-and-authorization/secure-web-applications-with-managed-authentication.md]: Resolved with [Address comments](https://github.com/wso2/docs-choreo-dev/pull/1542/commits/fe955e7e298df1cd4770bf6e2158341b6ef6eed4)

File: en/docs/authentication-and-authorization/secure-web-applications-with-managed-authentication.md (hunk 3 above)
Comment (ThumulaPerera) [reply en/docs/authentication-and-authorization/secure-web-applications-with-managed-authentication.md]: Resolved with [Address comments](https://github.com/wso2/docs-choreo-dev/pull/1542/commits/7aa96f4c5ddc4c4b9108fb3d695874675d092f1d)

File: en/docs/authentication-and-authorization/secure-web-applications-with-managed-authentication.md [hunk 4]
```diff
@@ -115,53 +115,21 @@ If you enable Choreo's managed authentication, you don't have to manually add an
 If Choreo's managed authentication is disabled, you must ensure that your web application attaches a valid access token to the API call.
 
 
-### Step 1.5: Refresh session
+### Step 1.5: Handle session expiry
 
-The logged is user's session expires when the access token provided by the identity provider you have configured expires. After the session expires, the session must be refreshed to successfully call a Choreo-deployed API. For session refresh to function, you must set up the identity provider to issue a refresh token in addition to the access token.
+The logged in user's session expires when the time configured for `Session Expiry Time` is exceeded. `401 Unauthorized` response status code for a Choreo API request from a logged in user with the necessary permissions indicates a possible session expiry. After the session expires, the user should re-login.
```
Comment (Nashaath) [inline en/docs/authentication-and-authorization/secure-web-applications-with-managed-authentication.md:120]: Suggestion
When a user session exceeds the configured session expiry time, it automatically expires. A `401 Unauthorized` response status code for a Choreo API request from a logged-in user indicates that the session may have expired, requiring the user to re-login.

File: en/docs/authentication-and-authorization/secure-web-applications-with-managed-authentication.md [hunk 5]
```diff
@@ -115,53 +115,21 @@ If you enable Choreo's managed authentication, you don't have to manually add an
...
-``` javascript
-const response = await fetch("/auth/refresh", { method: "POST" })
-```
-
-If the refresh token is valid, the session will be refreshed, and the system will respond with a `204 No Content` status.
-
-If the refresh token has expired, the session will not be refreshed, and the system will respond with a `401 Unauthorized` status.
-
-To automate the session refresh process on receiving a `401 unauthorized` response from the Choreo API, you can encapsulate the requests with the refresh logic. The following is an example code snippet that wraps GET requests:
+To programmatically re-login on receiving a `401 unauthorized` response from a Choreo API, you can encapsulate the request with the re-login logic. The following is an example code snippet that wraps GET requests:
```
Comment (Nashaath) [inline en/docs/authentication-and-authorization/secure-web-applications-with-managed-authentication.md:122]: Suggestion:
To programmatically handle session expiry and automatically re-login upon receiving a `401 Unauthorized` response from a Choreo API, you can encapsulate the request with re-login logic. The following sample code snippet shows how to wrap GET requests:

Comment (ThumulaPerera) [reply en/docs/authentication-and-authorization/secure-web-applications-with-managed-authentication.md]: Resolved with [Incorporate suggestions](https://github.com/wso2/docs-choreo-dev/pull/1542/commits/15d539a1821850398bdf58d6114b7f198fe47e50)

File: en/docs/authentication-and-authorization/secure-web-applications-with-managed-authentication.md (hunk 4 above)
Comment (ThumulaPerera) [reply en/docs/authentication-and-authorization/secure-web-applications-with-managed-authentication.md]: Resolved with [Incorporate suggestions](https://github.com/wso2/docs-choreo-dev/pull/1542/commits/15d539a1821850398bdf58d6114b7f198fe47e50)

**Review by Nashaath — APPROVED** [review]

**Pull request merged**